lms/djangoapps/grades/tasks.py

```diff
@@ -23,7 +23,9 @@
 
 
 @task(default_retry_delay=30, routing_key=settings.RECALCULATE_GRADES_ROUTING_KEY)
-def recalculate_subsection_grade(user_id, course_id, usage_id, only_if_higher, raw_earned, raw_possible, **kwargs):
+def recalculate_subsection_grade(
```

Comment:
Yeah - I think the grades/models.py events are run in the celery process.  It's possible that this issue wouldn't be caught by your integration test, since all celery work is done synchronously and in the same process by the python tests.